### PR TITLE
feat(examples): adyen payments integration

### PR DESCRIPTION
**Pull request by mkucmus**
### Description

the PR shows how to integrate Adyen drop-in component working with Shopware 6 using Composable Frontends.

closes: #406 

- adds a nuxt plugin that exposes the adyenCheckout instance factory
- adds a component to display a drop-in by Adyen
- adds an example logic to communicate between frontend and Shopware 6 instance

### TODO
- [x] add interaction with sw6 via store-api
- [x] consider more dynamic configuration of adyen checkout -> move from module options to composable function
- [x] add README.md
- [x] add styling

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Screenshots (if applicable)

image

### CODE REVIEW

**Review by patzick — COMMENTED** [review]
Good work here, let's just change tailwind to UnoCSS for consistency :)

**Comment by mkucmus** [comment]
> Good work here, let's just change tailwind to UnoCSS for consistency :)

sure, I made it consistent now 🙌🏼

File: examples/adyen-dropin-component/app.vue [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+// composable used to add to cart a product
+const { addProduct } = useCart();
+// configured apiClient instance in order to make a custom API call
+const { apiInstance } = useShopwareContext();
+// current step for progress bar displaying <ProgressBar /> in a template
+const activeStep = ref(0);
+// container for keeping and displaying an incoming payment state from adyen
+const paymentState = ref();
+
+try {
```
Comment (mdanilowicz) [inline examples/adyen-dropin-component/app.vue:24]: It is not good practice to have so many actions in single try/catch block, also console error can be mislead if errors comes `addProduct`.

File: examples/adyen-dropin-component/app.vue [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    orderId: order.id,
+    finishUrl: "http://localhost:3000/success",
+    errorUrl: "http://localhost:3000/failure?payment-failed",
+    // adyen specific data
+    stateData: JSON.stringify(state.data),
+  });
+  activeStep.value = 4;
+
+  // check for current payment status after payment authorization attempt
+  const paymentStateResponse = await apiInstance.invoke.post(
```
Comment (mdanilowicz) [inline examples/adyen-dropin-component/app.vue:65]: Maybe we can add this as a composable and use it in this example? It will be more reusable.

**Review by mdanilowicz — CHANGES_REQUESTED** [review]

File: examples/adyen-dropin-component/app.vue (hunk 1 above)
Comment (mkucmus) [reply examples/adyen-dropin-component/app.vue:24]: good point, would be enough to just pass error object to console.error? the example should be as short as possible 👀

File: examples/adyen-dropin-component/app.vue (hunk 2 above)
Comment (mkucmus) [reply examples/adyen-dropin-component/app.vue:65]: good point, we already have such method in useOrder composable, but the example is for extraordinary situation when some extension like "adyen" adds some handling not in shopware's manner. so in this case, we could use our composable if only the parameters passed would be in a different key. 

the value of the example is how to deal with unexpected situations like this one.

**Review by mdanilowicz — APPROVED** [review]

**Pull request merged**